### PR TITLE
Remove Tag from Hazard

### PR DESCRIPTION
**Pull request by emanuel-schmid**
Changes proposed in this PR:
- removes tag attribute from Hazard and subclasses

This PR fixes #

### PR Author Checklist

- [ ] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [x] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [x] [Tests][testing] updated
- [x] [Tests][testing] passing
- [x] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

**Comment by chahank** [comment]
Looks good to me! Can you please also update the changelog?

File: climada/hazard/base.py [hunk 1]
```diff
@@ -1887,21 +1871,11 @@ def from_hdf5(cls, file_name):
         hazard_kwargs = dict()
         with h5py.File(file_name, 'r') as hf_data:
             for (var_name, var_val) in haz.__dict__.items():
-                if var_name != 'tag' and var_name not in hf_data.keys():
+                if var_name not in hf_data.keys():
                     continue
                 if var_name == 'centroids':
                     hazard_kwargs["centroids"] = Centroids.from_hdf5(
                         hf_data.get(var_name))
-                elif var_name == 'tag':
```
Comment (chahank) [inline climada/hazard/base.py:1895]: Will this not make it impossible to read files created with previous versions of CLIMADA as the `haz_type` cannot be loaded?

File: climada/hazard/base.py [hunk 2]
```diff
@@ -1887,21 +1871,11 @@ def from_hdf5(cls, file_name):
...
-                    # legacy behavior: haz_type used to be part of the hazard tag
-                    if "haz_type" in hf_data.keys():
```
Comment (chahank) [inline climada/hazard/base.py:1897]: Will this not make it impossible to read files created with previous versions of CLIMADA as the haz_type cannot be loaded?

File: climada/hazard/base.py [hunk 3]
```diff
@@ -2183,7 +2157,6 @@ def _read_att_mat(data, file_name, var_names, centroids, description):
                 file_name, data[var_names['var_name']['ev_name']])
         except KeyError:
             attrs["event_name"] = list(attrs["event_id"])
-        attrs["description"] = description
         try:
             comment = u_hdf5.get_string(data[var_names['var_name']['comment']])
             attrs["description"] += ' ' + comment
```
Comment (chahank) [inline climada/hazard/base.py:2162]: What is this attribute description? Is this not part of the tag?

File: script/jenkins/petals_regression_test/Jenkinsfile [hunk 4]
```diff
@@ -14,7 +14,7 @@ pipeline {
         cp ~/jobs/petals_install_env/workspace/climada.conf climada.conf
         python script/jenkins/set_config.py test_directory ~/jobs/petals_install_env/workspace/climada_petals
 
-        pytest --junitxml=tests_xml/tests.xml ~/jobs/petals_install_env/workspace/climada_petals
+        PYTHONPATH=.:$PYTHONPATH pytest --junitxml=tests_xml/tests.xml ~/jobs/petals_install_env/workspace/climada_petals
```
Comment (chahank) [inline script/jenkins/petals_regression_test/Jenkinsfile:17]: Part of this PR?

Comment (peanutfun) [reply script/jenkins/petals_regression_test/Jenkinsfile:17]: If you need the current directory in your PYTHONPATH, call pytest with the python interpreter like so:
```
python -m pytest
```
See [Calling pytest through python -m pytest](https://docs.pytest.org/en/7.1.x/how-to/usage.html#calling-pytest-through-python-m-pytest)

However, this should not be necessary if both modules are installed

Comment (emanuel-schmid) [reply script/jenkins/petals_regression_test/Jenkinsfile:17]: No, not part of the PR, that is also in develop and right that's a substitute for `python -m pytest` (thanks @peanutfun for pointing this out, didn't know that)
And it's necessary, exactly _because_ both modules are installed. 😁

File: climada/hazard/base.py (hunk 2 above)
Comment (emanuel-schmid) [reply climada/hazard/base.py:1897]: Good question! And if so it should show in the integration test, shouldn't it?

File: climada/hazard/base.py (hunk 3 above)
Comment (emanuel-schmid) [reply climada/hazard/base.py:2162]: 👍 I'll remove it as well.

File: climada/hazard/base.py (hunk 2 above)
Comment (chahank) [reply climada/hazard/base.py:1897]: If the integration tests would be complete yes. But they are not. Ideally, we should write one now. (actually a unit test).

File: script/jenkins/petals_regression_test/Jenkinsfile (hunk 4 above)
Comment (peanutfun) [reply script/jenkins/petals_regression_test/Jenkinsfile:17]: I don't understand. If both are installed, the interpreter should have no trouble finding them. What are the contents of the `./` directory and why do you need to add it to the PYTHONPATH?

Comment (emanuel-schmid) [reply script/jenkins/petals_regression_test/Jenkinsfile:17]: './' is the root directory of the repository. Putting it in the first place of the python path is meant to make sure any climada module is taken from here rather then from the climada_petals installation.
Without this `import climada` yields the climada module from the develop branch.
Note that the environment is not installed but just activated.

File: climada/hazard/base.py (hunk 2 above)
Comment (emanuel-schmid) [reply climada/hazard/base.py:1897]: Actually it's not a problem: the hazard type is read below in lines
```python
    elif isinstance(var_val, np.ndarray) and var_val.ndim == 1:
        hazard_kwargs[var_name] = np.array(hf_data.get(var_name))
```

Comment (emanuel-schmid) [reply climada/hazard/base.py:1897]: The test for it is `climada.test.test_api_client.TestClient.test_get_hazard`.

File: script/jenkins/petals_regression_test/Jenkinsfile (hunk 4 above)
Comment (peanutfun) [reply script/jenkins/petals_regression_test/Jenkinsfile]: I think I understand. Installing Petals will pull the pip version of Climada, but you want the local one, right? Then I suggest to call pytest from the interpreter, as discussed. Setting the Pythonpath manually seems like a "dirtier" option to me.

These changes are now gone somehow?

File: CHANGELOG.md [hunk 5]
```diff
@@ -83,6 +83,7 @@ Removed:
 - `Centroids.set_raster_from_pix_bounds` [#721](https://github.com/CLIMADA-project/climada_python/pull/721)
 - `requirements/env_developer.yml` environment specs. Use 'extra' requirements when installing the Python package instead [#712](https://github.com/CLIMADA-project/climada_python/pull/712)
 - `Impact.tag` attribute. This change is not backwards-compatible with respect to the files written and read by the `Impact` class [#743](https://github.com/CLIMADA-project/climada_python/pull/743)
+- `Hazard.tag` attribute. This change is not backwards-compatible with respect to the files written and read by the `Hazard` class [#767](https://github.com/CLIMADA-project/climada_python/pull/767)
```
Comment (peanutfun) [inline CHANGELOG.md:86]: I think we should combine all these changelogs into a single log once we are finished and before the next large release

**Review by peanutfun — APPROVED** [review]
Looks good to me, nice work! What about the failing compatibility test? The adjustment on the Pythonpath is gone from the changes, somehow.

**Pull request merged**